File: app/src/main/java/com/example/mike/mp3player/client/views/buttons/MediaButton.java
package com.example.mike.mp3player.client.views.buttons;

import android.content.Context;
import android.os.Handler;
import android.widget.ImageView;

import androidx.annotation.NonNull;

import com.example.mike.mp3player.client.MediaControllerAdapter;

import javax.inject.Named;

public abstract class MediaButton {


    protected final Context context;
    protected final MediaControllerAdapter mediaControllerAdapter;
    protected final Handler mainUpdater;
    protected ImageView view;

    protected MediaButton(Context context, @NonNull MediaControllerAdapter mediaControllerAdapter,
                           @Named("main") Handler mainUpdater) {
        this.context = context;
        this.mediaControllerAdapter = mediaControllerAdapter;
        this.mainUpdater = mainUpdater;
    }
}

File: client/src/main/java/com/github/goldy1992/mp3player/client/views/buttons/MediaButton.kt
package com.github.goldy1992.mp3player.client.views.buttons

import android.content.Context
import android.view.View
import android.widget.ImageView
import androidx.annotation.DrawableRes
import com.github.goldy1992.mp3player.client.MediaControllerAdapter
import com.github.goldy1992.mp3player.commons.Constants
import com.github.goldy1992.mp3player.commons.LogTagger
import com.google.android.material.button.MaterialButton

abstract class MediaButton protected constructor(protected val context: Context,
                                                 protected val mediaControllerAdapter: MediaControllerAdapter) : LogTagger {
    protected lateinit var view: MaterialButton

    open fun init(imageView: MaterialButton) {
        view = imageView
        view.setOnClickListener(::onClick)
    }

    abstract fun onClick(view: View?)

    protected fun setImage(@DrawableRes drawableRes: Int) {
        this.setImage(drawableRes, Constants.OPAQUE)
    }

    protected fun setImage(@DrawableRes drawableRes: Int, alpha: Int) {
        val drawable = context.getDrawable(drawableRes)
        view.icon = drawable
        view.icon.alpha = alpha
    }

}